config/plugins/barefoot.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
+
+import click
+import json
+import subprocess
+from sonic_py_common import device_info
+
+def abort_if_false(ctx, param, value):
+    if not value:
+        ctx.abort()
+
+@click.group()
+def barefoot():
+    pass
+
+@barefoot.command()
+@click.option('-y', '--yes', is_flag=True, callback=abort_if_false,
+    expose_value=False, prompt='Current config will be cleared and config will '
+    'be reloaded from the default config file, continue?')
+@click.argument('profile')
+def profile(profile):
+    # Check if profile can be changed
+    completed_process = subprocess.run(['docker', 'exec', '-it', 'syncd',
+        'test', '-h', '/opt/bfn/install'])
+    if completed_process.returncode != 0:
+        click.echo('Cannot change profile: default one is in use')
+        raise click.Abort()
+    
+    # Get chip family
+    hwsku_dir = device_info.get_path_to_hwsku_dir()
+    with open(hwsku_dir + '/switch-tna-sai.conf') as file:
+        chip_family = json.load(file)['chip_list'][0]['chip_family']
+    
+    # Check if profile is supported
+    if chip_family == 'Tofino' and profile[0] == 'y' or \
+        chip_family == 'Tofino2' and profile[0] == 'x':
+        click.echo('Specified profile is unsupported on the system')
+        raise click.Abort()
+    
+    # Check if profile exists
+    completed_process = subprocess.run(['docker', 'exec', '-it', 'syncd',
+        'test', '-d', '/opt/bfn/install_' + profile + '_profile'])
+    if completed_process.returncode != 0:
+        click.echo('No profile with the provided name found')
+        raise click.Abort()
+    
+    # Parse config, add profile and write back
```

Comment:
># Parse config, add profile and write back [](http://example.com/codeflow?start=4&length=42)

The normal `config` subcommand will only update Redis ConfigDB content. And users are expected to run `config save` to save the Redis ConfigDB content to `/etc/sonic/config_db.json`.

Could you follow the same flow?

Reply:
Thanks for your review.
Now, the `config` subcommand only updates Redis ConfigDB content without modifying `/etc/sonic/config_db.json`.